src/faaslet/Faaslet.cpp

```diff
@@ -213,16 +209,12 @@ namespace faaslet {
 
             // Check if we need to restore from a different snapshot
             const std::string snapshotKey = msg.snapshotkey();
-#if(FAASM_SGX == 1)
-            if(!snapshotKey.empty() && msg.function().find(FAASM_SGX_FUNC_SUFFIX,0) == std::string::npos){
-#else
-            if(!snapshotKey.empty()){
-#endif
+            if (!snapshotKey.empty() && !msg.issgx()) {
```

Comment:
I've added this new `issgx` field to the incoming messages. 

This is how users can request to run their functions with or without SGX, rather than relying on the function suffix